openwisp_monitoring/device/tasks.py

```diff
@@ -28,7 +29,11 @@ def trigger_device_checks(pk, recovery=True):
     except ObjectDoesNotExist:
         logger.warning(f'The device with uuid {pk} has been deleted')
         return
-    check_ids = list(device.checks.filter(is_active=True).values_list('id', flat=True))
+    check_ids = list(
+        device.checks.filter(
+            is_active=True, check_type__in=device.monitoring.get_critical_checks()
```

Comment:
Does this query trigger the index use too?

Reply:
Yes, I did the analysis for this query. 

```sql
Index Scan using check_check_content_b49654_idx on check_check
  Index Cond: (content_type_id = 24 AND object_id = '...' AND is_active = true)
  Filter: (check_type IN (...))
```